cms/topics/tests/test_viewsets.py

```diff
@@ -0,0 +1,217 @@
+
+    def test_article_choose_view_with_topic_filter(self):
+        """Test the article chooser view with topic_page_id filter."""
+        response = self.client.get(f"{self.article_chooser_url}?topic_page_id={self.topic_page.id}")
+
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "wagtailadmin/generic/chooser/chooser.html")
+
+        # Check articles are shown
+        self.assertListEqual(
+            response.context["items"].object_list,
+            [self.second_article, self.article],
+        )
+
+        # Check column headers
+        self.assertContains(response, "Release date")
+        self.assertContains(response, "Status")
+
+    def test_article_choose_view_without_topic_filter(self):
+        """Test the article chooser view without topic_page_id returns no results."""
+        response = self.client.get(self.article_chooser_url)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "wagtailadmin/generic/chooser/chooser.html")
+        self.assertNotContains(response, "Article 1")
+        self.assertNotContains(response, "Article 2")
+
+    def test_article_results_view_with_topic_filter(self):
+        """Test the article AJAX results view with topic_page_id filter."""
+        response = self.client.get(f"{self.article_chooser_results_url}?topic_page_id={self.topic_page.id}&q=Another")
+
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "wagtailadmin/generic/chooser/results.html")
+
+        # Check articles are ordered by release_date descending
+        self.assertListEqual(
+            response.context["items"].object_list,
+            [self.second_article],
+        )
+
+    def test_methodology_choose_view_with_topic_filter(self):
+        """Test the methodology chooser view with topic_page_id filter."""
+        response = self.client.get(f"{self.methodology_chooser_url}?topic_page_id={self.topic_page.id}")
+
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "wagtailadmin/generic/chooser/chooser.html")
+
+        # Check methodology is shown
+        self.assertListEqual(
+            response.context["items"].object_list,
+            [self.methodology, self.second_methodology],
+        )
+
+    def test_methodology_results_view_with_topic_filter(self):
+        """Test the methodology AJAX results view with topic_page_id filter."""
+        response = self.client.get(
+            f"{self.methodology_chooser_results_url}?topic_page_id={self.topic_page.id}&q=Another"
+        )
+
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "wagtailadmin/generic/chooser/results.html")
+
+        self.assertListEqual(
+            response.context["items"].object_list,
+            [self.second_methodology],
+        )
+
+
+class ViewSetConfigurationTest(TestCase):
```

Comment:
Question: Why are these in a separate test case rather than being included in the ones above? It feels like they'd be better grouped with the tests for how the view works itself.

Reply:
initially I had a view that this would be more extensive, but in the end it was not needed, so it kind of stayed. merged in the test cases above (069f8cd), thanks